x-pack/platform/packages/shared/kbn-inference-cli/README.md

```diff
@@ -0,0 +1,40 @@
+# @kbn/inference-cli
+
+Exposes an Inference (plugin) API client for scripts, that mimicks the `chatComplete`
+and `output` APIs that are available on its start contract. It depends on the KibanaClient
+that is exposed from the `@kbn/kibana-api-cli` package. It automatically selects a
+connector if available. Usage:
+
+```ts
+import { createInferenceClient } from '@kbn/inference-cli';
+import { ToolingLog } from '@kbn/tooling-log';
+
+const log = new ToolingLog();
+const inferenceClient = await createInferenceClient({
+  log,
+  // pass in a signal that is triggered on teardown
+  signal: new AbortController().signal,
+});
+
+const response = await inferenceClient.output({
+  id: 'extract_personal_details',
+  input: `Sarah is a 29-year-old software developer living in San Francisco.`,
+  schema: {
+    type: 'object',
+    properties: {
+      name: { type: 'string' },
+      age: { type: 'number' },
+      city: { type: 'string' },
+    },
+    required: ['name'],
+  } as const,
+});
+
+log.info(response.output);
+```
+
+Running a recipe:
+
+```
+$ yarn run ts-node x-pack/solutions/observability/packages/kbn-genai-cli/recipes/hello_world.ts
```

Comment:
Add `ts-node --transpileOnly` if you want to bypass compile errors:

```suggestion
$ yarn run ts-node --transpileOnly x-pack/solutions/observability/packages/kbn-genai-cli/recipes/hello_world.ts
```

I was getting a silly error that shouldn't prevent the script from executing (I need to use the enum I guess) but typescript stalls on compile errors:

<img width="1548" alt="image" src="https://github.com/user-attachments/assets/2592648d-ee6a-4fe9-b5cc-49658f516f3f" />

Reply:
for the latter, I'm not a fan of using scripts that way tbh. I have a function defined in my zshrc that prepends the called script with the babel-register stuff, that's probably a better way to do it but I also don't want to be too prescriptive here.